test/functional/feature_evm_token_split.py

```diff
@@ -0,0 +1,768 @@
+            assert_equal(
+                self.intrinsics_contract.functions.dvmBlockCount().call(),
+                self.nodes[0].getblockcount(),
+            )
+
+    def fund_address(self, source, destination, amount=20):
+
+        # Fund address with META
+        self.nodes[0].transferdomain(
+            [
+                {
+                    "src": {"address": source, "amount": f"{amount}@META", "domain": 2},
+                    "dst": {
+                        "address": destination,
+                        "amount": f"{amount}@META",
+                        "domain": 3,
+                    },
+                    "singlekeycheck": False,
+                }
+            ]
+        )
+        self.nodes[0].generate(1)
+
+        # Fund address with DFI
+        self.nodes[0].transferdomain(
+            [
+                {
+                    "src": {"address": source, "amount": "1@DFI", "domain": 2},
+                    "dst": {
+                        "address": destination,
+                        "amount": "1@DFI",
+                        "domain": 3,
+                    },
+                    "singlekeycheck": False,
+                }
+            ]
+        )
+        self.nodes[0].generate(1)
+
+        # Check META balance
+        meta_contract = self.nodes[0].w3.eth.contract(
+            address=self.contract_address_metav1, abi=self.dst20_abi
+        )
+        assert_equal(
+            meta_contract.functions.balanceOf(destination).call()
+            / math.pow(10, meta_contract.functions.decimals().call()),
+            Decimal(amount),
+        )
+
+        # Check DFI balance
+        balance = self.nodes[0].eth_getBalance(destination)
+        assert_equal(balance, int_to_eth_u256(1))
+
+    def transfer_domain_split(self):
+
+        # Rollback
+        self.rollback_to(self.block_height)
+
+        # Fund address
+        self.fund_address(self.address, self.evm_address)
+
+        # Split token
+        self.split_token(self.contract_address_metav1, self.contract_address_metav2)
+
+        # Generate new destination address
+        destination_address = self.nodes[0].getnewaddress("", "legacy")
+
+        # Transfer out METAv1 tokens
+        self.nodes[0].transferdomain(
+            [
+                {
+                    "src": {
+                        "address": self.evm_address,
+                        "amount": "20@META/v1",
+                        "domain": 3,
+                    },
+                    "dst": {
+                        "address": destination_address,
+                        "amount": "20@META/v1",
+                        "domain": 2,
+                    },
+                    "singlekeycheck": False,
+                }
+            ]
+        )
+        self.nodes[0].generate(1)
+
+        # Check META balance
+        assert_equal(
+            self.nodes[0].getaccount(destination_address)[0], "40.00000000@META"
+        )
```

Comment:
i would also add a check that there was no META/v1 added on DVM side. for safety sake.

also check please check that the gov-economy values for TD (transfered to EVM and transfered back) are updated. should have the META/v1 reduced now.